pkg/commands/catalog/mirror/mirror.go

```diff
@@ -104,6 +114,52 @@ func Mirror(options Options) error {
 		}
 	}
 
+	if options.Download {
+		// Create a temporary directory to place the oci-archive files
+		ociArchiveDir, err := os.MkdirTemp("", "oci-archive")
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(ociArchiveDir)
+		counter := 1
+		for _, image := range images {
+			imageInfo, err := imageUtil.SplitImage(image)
+			if err != nil {
+				return err
+			}
+			log.Debugf("Copying %s:%s to system", imageInfo.BaseImage, imageInfo.Tag)
+			timeoutHappened := true
+			for i := 0; i < 5; i++ {
+				err = imageUtil.Copy(fmt.Sprintf("docker://%s", image), "oci-archive:"+ociArchiveDir+"/"+strconv.Itoa(counter)+".oci:"+imageInfo.BaseImage+":"+imageInfo.Tag, "", copy.CopyAllImages)
+				if err == nil {
+					timeoutHappened = false
+					break
+				} else if !strings.Contains(err.Error(), "500 Internal Server Error") {
+					return err
+				} else {
+					// Delete oci-archive file and backoff
+					os.Remove(ociArchiveDir + "/" + strconv.Itoa(counter) + ".oci")
+					log.Debugf("Backing off and retrying pulling %s:%s from the registry", imageInfo.BaseImage, imageInfo.Tag)
+					time.Sleep(30 * time.Second)
```

Comment:
30 secs this is too long.  Wait a few secs (2 or 3)

Reply:
Ok!